src/pages/Flow/UpcomingRuns-Tile.vue

```diff
@@ -41,86 +36,82 @@ export default {
-      })
-    },
     loading() {
       return this.loadingKey > 0
     },
     paused() {
       return this.tenant?.settings?.work_queue_paused
     },
+    lateRuns() {
+      if (!this.upcomingFlowRunsData) return null
+
+      return this.upcomingFlowRunsData.filter(run => this.runIsLate(run))
+    },
     upcomingRuns() {
-      if (!this.upcoming) return null
-      return this.upcoming.filter(run => {
-        return this.getTimeOverdue(run.scheduled_start_time) <= 20000
-      })
+      if (!this.upcomingFlowRunsData) return null
+
+      return this.upcomingFlowRunsData.filter(run => !this.runIsLate(run))
     },
     title() {
-      if (!this.upcoming) return
-      let title = ''
+      if (!this.upcomingFlowRunsData) return null
 
-      if (this.tab == 'upcoming') {
-        title =
-          this.loading > 0
-            ? 'Upcoming runs'
-            : `${this.upcomingRuns?.length || 0} upcoming runs`
-      }
-
-      if (this.tab == 'late') {
-        title = this.loading
-          ? 'Late runs'
-          : `${this.lateRuns?.length +
-              (this.lateRuns?.length >= 1000 ? '+' : '') || 0} late runs`
-      }
-
-      return title
+      return this.tabProperties[this.tab].title
     },
     titleIcon() {
-      let icon = ''
-
-      if (this.tab == 'upcoming') {
-        icon = 'access_time'
-      }
+      return this.tabProperties[this.tab].icon
+    },
+    titleIconColor() {
+      if (this.loading) return 'grey'
 
-      if (this.tab == 'late') {
-        icon = 'timelapse'
-      }
+      return this.tabProperties[this.tab].icon_color
+    },
+    systemBarColor() {
+      if (this.loading || !this.upcomingFlowRunsData) return 'secondaryGray'
 
-      return icon
+      return this.lateRuns?.length > 0 ? 'deepRed' : 'Success'
     },
-    titleIconColor() {
-      return this.loading
-        ? 'grey'
-        : this.tab == 'upcoming'
-        ? 'primary'
-        : this.lateRuns?.length > 0
-        ? 'deepRed'
-        : 'Success'
+    tabProperties() {
+      var upcomingCount = this.upcomingRuns?.length || 0
+      var lateCount = this.lateRuns?.length || 0
+
+      return {
+        [this.tabs.upcoming]: {
+          title: `${upcomingCount} upcoming runs`,
+          icon: 'access_time',
+          icon_color: 'primary'
+        },
+        [this.tabs.late]: {
+          title: `${lateCount > 999 ? '1,000+' : lateCount} late runs`,
+          icon: 'timelapse',
+          icon_color: lateCount > 0 ? 'deepRed' : 'Success'
+        }
+      }
     }
   },
   watch: {
-    upcoming(val) {
-      if (!val) return
-      if (this.lateRuns?.length > 0) {
-        this.tab = 'late'
-      }
-      if (this.lateRuns?.length <= 0) {
-        this.tab = 'upcoming'
+    upcomingFlowRunsData(val) {
+      if (!val || !this.pristine) return
+
+      if (this.tab == this.tabs.upcoming && this.lateRuns?.length > 0) {
+        this.tab = this.tabs.late
```

Comment:
quick note here, I was frustrated that if there are runs in the "late" tab, even when I explicitly go back to the "upcoming" tab the system would continue to kick me back to "late" on a regular interval. I change this behavior to only programmatically switch the user to "late" tab if the user hasn't explicitly changed tabs